src/analyses/libraryDsl.ml

```diff
@@ -0,0 +1,104 @@
+open LibraryDesc
+
+(** First-class patterns for arguments matching.
+    @see <https://github.com/ocaml-ppx/ppxlib/blob/main/src/ast_pattern.ml> for inspiration from ppxlib. *)
+module Pattern =
+struct
```

Comment:
The intention of the `Pattern` module is a bit hard to understand at first sight, and the link does not really offer an explanation. Adding an additional comment might be useful.

Reply:
True, although this module is completely internal to the implementation. `libraryDsl.mli` doesn't expose it at all and documents everything that is intended to be used. Also there's additional `docs/` about it all.